build.rs

```diff
@@ -271,8 +275,78 @@ const MSVC: &'static str = "msvc";
 const MSVC_OBJ_OPT: &'static str = "/Fo";
 const MSVC_OBJ_EXT: &'static str = "obj";
 
+const GENERATED_VERSIONED_HEADER: &str = "generated_versioned.h";
+const GENERATED_YASM_VERSIONED_HEADER: &str = "yasm_versioned.h";
+const GENERATED_VERSIONED_MACRO: &str = "VERSIONED";
+const GENERATED_INCLUDE_DIR: &str = "generated_headers";
+
+fn generated_include_dir() -> PathBuf {
+    PathBuf::from(::std::env::var("OUT_DIR").unwrap()).join(GENERATED_INCLUDE_DIR)
+}
+
+fn custom_versioned_symbols_header() -> PathBuf {
+    PathBuf::from(env!("CARGO_MANIFEST_DIR")).join("include").join("versioned.h")
+}
+
+fn yasm_versioned_symbols_header() -> PathBuf {
+    PathBuf::from(::std::env::var("OUT_DIR").unwrap())
+        .join(GENERATED_INCLUDE_DIR)
+        .join(GENERATED_YASM_VERSIONED_HEADER)
+}
+
+// Take the versioned header in `from`, replace #define with %define, preprocess
+// it with the C compiler, write it out to `yasm_versioned_symbols_header()`.
+fn write_yasm_versioned_header(from: &Path) -> io::Result<()> {
+    use std::fs::File;
+    use std::io::{BufReader, BufRead, Write, Cursor};
+
+    // Create the data for the intermediate file.
+    let mut new_data = String::new();
+    for line in BufReader::new(File::open(from)?).lines() {
+        use std::fmt::Write;
+        let line = line?;
+        if line.starts_with("#define") {
+            write!(new_data, "%{}\n", &line[1..]).expect("write to str");
```

Comment:
Why is this needed? Notice that none of the *.pl files do this. I think it may be because PerlAsm does this translation itself. See the suggestion below about having PerlAsm write an `#include` into the generated asm files.

Reply:
This is for `yasm` on Windows.